File: leetcode2/1easy/염류선/Q2194.java
package Leetcode.염류선;

import java.util.ArrayList;
import java.util.List;

public class Q2194 {
    public List<String> cellsInRange(String s) {
        List<String> result = new ArrayList<>();

        char a = s.charAt(0);
        char b = s.charAt(3);
        int c = s.charAt(1) - '0';
        int d = s.charAt(4) - '0';

        for (char col = a; col <= b; col++) {
            for (int row = c; row <= d; row++) {
                result.add("" + col + row);
            }
        }

        return result;
    }
}

File: leetcode2/1easy/강병호/Q2194.py
class Solution:
    def cellsInRange(self, s: str) -> list[str]:
        # cells = list(map(str, s.split(":")))
        startCol = s[0]
        endCol = s[3]
        startRow = int(s[1])
        endRow = int(s[4])

        result = list()

        for col in range(ord(startCol), ord(endCol) + 1):
            for row in range(startRow, endRow+1):
                data = chr(col) + str(row)
                result.append(data)
        
        return result


if __name__ == "__main__":
    sol = Solution()
    print(sol.cellsInRange("K1:L2"))
    print(sol.cellsInRange("A1:F1"))